fix($parse): block assigning to fields of a constructor prototype

gkalpak opened the pull request:
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Bug fix.

**What is the current behavior? (You can also link to an open issue here)**
See #14939.

**Does this PR introduce a breaking change?**
No.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Other information**:
1. **1st commit:** Add a few more tests for constructor access.
2. **2nd commit:** Fix + tests.
3. **3rd commit:** Add missing error page.

gkalpak committed:
test($parse): test more constructors against `isecaf` error

gkalpak committed:
fix($parse): block assigning to fields of a constructor prototype

Fixes #14939

gkalpak committed:
docs($parse/isecaf): add missing error page

lgalfaso commented:
I find the tests a little bit too long (maybe it could be split into cases using `they`). One way or another, this is good to land. LGTM

gkalpak commented:
Thx, @lgalfaso! I'll break the tests up and merge.

gkalpak commented:
I split the tests up (and reverted the `isecac`-->`isecac` rename). I will merge once Travis is green.
Thx for the reviews, @lgalfaso and @petebacondarwin!

dcherman commented on src/ng/parse.js:
```diff
@@ -113,10 +113,28 @@ function ensureSafeFunction(obj, fullExpression) {
+        obj === functionConstructor.prototype) {
       throw $parseMinErr('isecaf',
-        'Assigning to a constructor is disallowed! Expression: {0}', fullExpression);
+        'Assigning to a constructor or it\'s prototype is disallowed! Expression: {0}',
```
`its` rather than `it's` :)

http://grammarist.com/spelling/its-its/

gkalpak replied:
Why is it that I can always find these typos in other people's PRs, but not mine :confused:
